influxql/ast.go

```diff
@@ -552,6 +554,31 @@ func (s *SetPasswordUserStatement) RequiredPrivileges() ExecutionPrivileges {
 	return ExecutionPrivileges{{Admin: true, Name: "", Privilege: AllPrivileges}}
 }
 
+type BackfillStatement struct {
+	QueryName string
+	Database  string
+	From      Expr
+	Until     Expr
+}
+
+func (b *BackfillStatement) String() string {
+	untilstr := ""
+	if b.Until != nil {
+		untilstr = fmt.Sprintf(" UNTIL %s", b.Until.String())
+	}
+	return fmt.Sprintf("BACKFILL %s ON %s FROM %s%s", b.QueryName, b.Database, b.From.String(), untilstr)
+}
+
+func (b *BackfillStatement) RequiredPrivileges() ExecutionPrivileges {
+	return []ExecutionPrivilege{
+		{
+			Admin:     false,
+			Name:      b.Database,
+			Privilege: WritePrivilege,
+		},
+	}
+}
```

Comment:
Problem with that is that we only have the CQ name at this point, not the CQ contents. In order to use the CQ being run, we need to look up the contents, which we don't have any handles to do at the moment.